src/app_service/service/message/service.nim

```diff
@@ -580,3 +592,35 @@ proc deleteMessage*(self: Service, messageId: string) =
 
   except Exception as e:
     error "error: ", methodName="deleteMessage", errName = e.name, errDesription = e.msg
+
+proc editMessage*(self: Service, messageId: string, updatedMsg: string) =
```

Comment:
@Khushboo-dev-cpp since we already have a method to convert a json to the appropriate `MessageDto`, let's use it. May we write this proc a bit shorter like:
```
proc editMessage*(self: Service, messageId: string, updatedMsg: string) =
  try:
    let response = status_go.editMessage(messageId, updatedMsg)

    var messagesArr: JsonNode
    var messages: seq[MessageDto]
    if(response.result.getProp("messages", messagesArr) and messagesArr.kind == JArray):    
      messages = map(messagesArr.getElems(), proc(x: JsonNode): MessageDto = x.toMessageDto())

    if(messages.len == 0):
      error "error: ", methodName="editMessage", errDesription = "messages array is empty"
      return

    let data = MessageEditedArgs(chatId: messages[0].chatId, message: messages[0])
    self.events.emit(SIGNAL_MESSAGE_EDITED, data)

  except Exception as e:
    error "error: ", methodName="editMessage", errName = e.name, errDesription = e.msg
```

You can add checks like this if you want before you emit a signal:
```
if messages[0].chatId.len == 0:
  error "error: ", methodName="editMessage", errDesription = "chat id is empty"
  return

if messages[0].editedAt <= 0:
  error "error: ", methodName="editMessage", errDesription = "message is not edited"
  return
```
to check chat id, and edited at, but that's not necessary in that context, cause those errors need to be handled on the `status-go` side.

Reply:
Done :D kept the check for editetAt just in case :)